blurring_as_a_service/inference_pipeline/submit_inference_pipeline.py

```diff
@@ -10,58 +12,90 @@
+
+        # Get the txt file that contains all paths of the files to run inference on
+        files_to_blur_path = customer['inputs']['files_to_blur'].format(
+            subscription=subscription_id,
+            resourcegroup=resource_group,
+            workspace=workspace_name,
+            datastore_name=f"{customer_name}_input_structured"
+        )
+
+        files_to_blur_txt = Input(
+            type=AssetTypes.URI_FILE,
+            path=files_to_blur_path,
+            description="Data to be blurred",
+        )
+
+        model_parameters = customer['model_parameters']
+        model_parameters_json = json.dumps(model_parameters)  # TODO it seems I can not pass a string to @command_component function
+
+        detect_and_blur_sensitive_data_step = detect_and_blur_sensitive_data(
+            mounted_root_folder=input_root_folder,
+            relative_paths_files_to_blur=files_to_blur_txt,
+            customer_name=customer_name,
+            model_parameters_json=model_parameters_json
+        )
+
+        azureml_outputs_formatted = customer['outputs']['results_path'].format(
+            subscription=subscription_id,
+            resourcegroup=resource_group,
+            workspace=workspace_name,
+            datastore_name=f"{customer_name}_output"
+        )
+
+        detect_and_blur_sensitive_data_step.outputs.results_path = Output(
+            type="uri_folder", mode="rw_mount", path=azureml_outputs_formatted
+        )
+        detect_and_blur_sensitive_data_step.outputs.yolo_yaml_path = Output(
+            type="uri_folder", mode="rw_mount", path=blob_container_path
+        )
 
     return {}
 
 
 def main():
-    aml_interface = AMLInterface()
-    settings = BlurringAsAServiceSettings.get_settings()
-
-    mounted_root_folder = Input(
-        type=AssetTypes.URI_FOLDER,
-        path=settings["inference_pipeline"]["inputs"]["root_folder"],
-        description="Data to be blurred",
-    )
-    relative_paths_files_to_blur = Input(
-        type=AssetTypes.URI_FILE,
-        path=settings["inference_pipeline"]["inputs"]["files_to_blur"],
-        description="Data to be blurred",
-    )
-    model = Input(
-        type=AssetTypes.URI_FOLDER,
-        path=settings["inference_pipeline"]["inputs"]["model"],
-        description="Model to use for the blurring",
-    )
-    inference_pipeline_job = inference_pipeline(
-        mounted_root_folder=mounted_root_folder,
-        relative_paths_files_to_blur=relative_paths_files_to_blur,
-        model=model,
-    )
+    inference_pipeline_job = inference_pipeline()
 
     inference_pipeline_job.settings.default_compute = settings[
         "aml_experiment_details"
     ]["compute_name"]
+
     pipeline_job = aml_interface.submit_pipeline_job(
         pipeline_job=inference_pipeline_job, experiment_name="inference_pipeline"
     )
     aml_interface.wait_until_job_completes(pipeline_job.name)
 
 
 if __name__ == "__main__":
+    aml_interface = AMLInterface()
```

Comment:
If you move them here they become global. The only thing I would keep global is the settings.